content/en/blog/_posts/2021-05-04-writing-a-controller-for-pod-labels.md

```diff
@@ -0,0 +1,469 @@
+---
+layout: blog
+title: "Writing a Controller for Pod Labels"
+date: 2021-05-04
+slug: writing-a-controller-for-pod-labels
+---
+
+**Authors**: Arthur Busser (Padok)
+
+[Operators][what-is-an-operator] are proving to be an excellent solution to
+running stateful distributed applications in Kubernetes. Open source tools like
+the [Operator SDK][operator-sdk] provide ways to build reliable and maintainable
+operators, making it easier to extend Kubernetes and implement custom
+scheduling.
+
+Kubernetes operators run complex software inside your cluster. The open source
+community has already built [many operators][operatorhub] for distributed
+applications like Prometheus, Elasticsearch, or Argo CD. Even outside of
+open source, operators can help to bring new functionality to your Kubernetes
+cluster.
+
+An operator is a set of [custom resources][custom-resource-definitions] and a
+set of [controllers][controllers] that, when combined, run complex software
+inside your Kubernetes cluster. A controller watches for changes to specific
+resources in the Kubernetes API and reacts by creating, updating, or deleting
+resources.
+
+The Operator SDK is best suited for building dependable operators. Nonetheless,
+you can use it to write a single controller. This post will walk you through
```

Comment:
> dependable ... nonetheless ... single

this doesn't make sense to me

Reply:
Originally, I had written "fully-fledged" instead of "dependable", which contrasted with "single" down the line. However, "fully-fledged" was deemed to be too idiomatic and I replaced it with "dependable". I now see that the paragraph does not really make sense anymore. I am going to find a better way to phrase this.

Reply:
I've rephrased:
> The Operator SDK is best suited for building **fully-featured** operators. Nonetheless, you can use it to write a single controller.